### DISCUSSION TITLE
Issue using createContractEventFilter with NextJS app directory

### DISCUSSION THREAD

**Discussion by neokry** [wevm/viem#374] [General]
When using `createContractEventFilter` in NextJS app directory `getFilterLogs` occasionally throws with a `filter not found error`. Anyone else running into an issue like this? Not sure if its an issue with viem, alchemy, or next.

**Comment by tmm** [comment]
Mind putting together a minimal reproducible example? Happy to take a look.

**Reply by neokry** [reply]
Having issues creating an example on stackblitz. only seems to really be reproducable when I deploy on vercel and it always seems to error the first call after deployment (not sure how helpful this is). It seems to be a viem error though because switching back to ethers for the log filter works with no issues.

**Reply by neokry** [reply]
This is how the logs were being queried if that helps: https://github.com/neokry/builder-reasons/blob/e76a60424c0db70a292077fcbc9bdb0cc9b89b95/data/eth/proposal.ts

**Reply by jxom** [reply]
Hmm, I seem to be creating the filter and getting logs fine... Would you be able to create a minimal reproduction like mine? Below is my working example:

https://viem-repro-374.vercel.app/
https://github.com/jxom/viem-repro-374

**Comment by hammeiam** [comment]
You might need to specify a startBlock for the filter to work

**Reply by neokry** [reply]
yes using startBlock fixed one of my issues but now running into issues even with startBlock set
